bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/types/BArrayType.java

```diff
@@ -128,8 +128,32 @@ public boolean equals(Object obj) {
 
     @Override
     public String toString() {
-        StringBuilder sb = new StringBuilder(elementType.toString());
-        return size != -1 ? sb.append("[").append(size).append("]").toString() : sb.append("[]").toString();
+        StringBuilder sb = new StringBuilder();
+        sb.append(getTypeName());
+        sb.append(getSizeString());
+        BType element = elementType;
+        while (element instanceof BArrayType) {
+            sb.append(((BArrayType) element).getSizeString());
```

Comment:
This cast can be moved to a variable as there are three casts happening for the same.

Reply:
Yes. changed into a variable in [a9602ad](https://github.com/ballerina-platform/ballerina-lang/pull/22708/commits/a9602ad8dfa8420eea723f0f163906c6dcfb1db2)